sdk/src/wallet/utxo.ts

```diff
@@ -69,36 +70,48 @@ export async function createBitcoinPsbt(
     const addressInfo = getAddressInfo(fromAddress);
 
     // TODO: possibly, allow other strategies to be passed to this function
-    const utxoSelectionStrategy = 'default';
+    const utxoSelectionStrategy: 'all' | 'default' = 'default';
 
     if (addressInfo.network === 'regtest') {
         throw new Error('Bitcoin regtest not supported');
     }
 
     // We need the public key to generate the redeem and witness script to spend the scripts
-    if (addressInfo.type === (AddressType.p2sh || AddressType.p2wsh)) {
+    if (
+        addressInfo.type === AddressType.p2sh ||
+        addressInfo.type === AddressType.p2wsh ||
+        addressInfo.type === AddressType.p2tr
+    ) {
         if (!publicKey) {
             throw new Error('Public key is required to spend from the selected address type');
         }
     }
 
     const esploraClient = new EsploraClient(addressInfo.network);
+    const ordinalsClient = new OrdinalsClient(addressInfo.network);
 
     let confirmedUtxos: UTXO[] = [];
+    let outputsS: OutputJson[] = [];
 
     if (feeRate) {
-        confirmedUtxos = await esploraClient.getAddressUtxos(fromAddress);
+        [confirmedUtxos, outputsS] = await Promise.all([
+            esploraClient.getAddressUtxos(fromAddress),
+            ordinalsClient.getInscriptionsFromAddress(fromAddress, 'cardinal'),
+        ]);
     } else {
-        [confirmedUtxos, feeRate] = await Promise.all([
+        [confirmedUtxos, feeRate, outputsS] = await Promise.all([
```

Comment:
Do we still need the `esploraClient.getAddressUtxos` if we already have the data from the ordinals api?

Also, another thing we should check before merging, is how the ord api handles utxos in the mempool. Ideally we want the behavior to be the same as the current esplora api

Reply:
We can not 100% sure about that if we do not simulate in mainnet, right? Currently checking their docs and tests to find a confirmation to that

Reply:
I would say the opposite - testing this on mainnet does not make us 100% sure. It's edge cases that I'm worried about. If we only have 1 data source, we don't need to worry about the edge case where one of the api updates before the other one does. E.g. if esplora updates more quickly than ord, then if an ordinal is sent to the address, esplora will find the utxo but ord does not have it marked as being an ordinal yet, which could lead to us spending it if we're not careful.

That's not to say that we can't keep 2 data sources, just that we need to be careful. I think the current approach _should_ be fine - we only use utxos that are reported as spendable by both esplora and ord